app/upgrades/v2/upgrades.go

```diff
@@ -64,68 +64,79 @@ func CreateUpgrade(includeAsyncICQ bool) upgrades.Upgrade {
+			// so we set allow list to empty and messages can be added later when needed via gov proposal
+			icaHostParams := icahosttypes.DefaultParams()
+			icaHostParams.AllowMessages = nil
+			if err := icaHostParams.Validate(); err != nil {
+				return nil, err
+			}
+			keepers.ICAHostKeeper.SetParams(sdkCtx, icaHostParams)
+
+			// update reward distribution events
+			err = UpdateRewardTrackerEventLastProcessedHeight(ctx, keepers.IncentiveKeeper)
+			if err != nil {
+				return nil, err
+			}
+
+			// Set the denom creation fee to ubbn
+			params := tokenfactorytypes.DefaultParams()
+			params.DenomCreationFee = sdk.NewCoins(sdk.NewInt64Coin(minttypes.DefaultBondDenom, 10_000_000))
+
+			if err := params.Validate(); err != nil {
+				return nil, err
+			}
+
+			if err := keepers.TokenFactoryKeeper.SetParams(ctx, params); err != nil {
+				return nil, err
+			}
+
+			return migrations, nil
 		}
-
-		sdkCtx := sdk.UnwrapSDKContext(ctx)
-
-		// Add a default rate limit to existing channels on port 'transfer'
-		if err := addRateLimits(sdkCtx, keepers.IBCKeeper.ChannelKeeper, keepers.RatelimitKeeper); err != nil {
-			return nil, err
-		}
-
-		// By default, ICQ allowed queries are empty. So no queries will be allowed until
-		// the allowed list is populated via gov proposal.
-		// For ICA host, by default all messages are allowed (using '*' wildcard),
-		// so we set allow list to empty and messages can be added later when needed via gov proposal
-		icaHostParams := icahosttypes.DefaultParams()
-		icaHostParams.AllowMessages = nil
-		if err := icaHostParams.Validate(); err != nil {
-			return nil, err
-		}
-		keepers.ICAHostKeeper.SetParams(sdkCtx, icaHostParams)
-
-		// update reward distribution events
-		err = UpdateRewardTrackerEventLastProcessedHeight(ctx, keepers.IncentiveKeeper)
-		if err != nil {
-			return nil, err
-		}
-
-		// Set the denom creation fee to ubbn
-		params := tokenfactorytypes.DefaultParams()
-		params.DenomCreationFee = sdk.NewCoins(sdk.NewInt64Coin(minttypes.DefaultBondDenom, 10_000_000))
-
-		if err := params.Validate(); err != nil {
-			return nil, err
-		}
-
-		if err := keepers.TokenFactoryKeeper.SetParams(ctx, params); err != nil {
-			return nil, err
-		}
-
-		return migrations, nil
 	}
 }
 
 // addRateLimits adds rate limit to the native denom ('ubbn') to all channels registered on the channel keeper
-func addRateLimits(ctx sdk.Context, chk transfertypes.ChannelKeeper, rlk ratelimitkeeper.Keeper) error {
+func addRateLimits(ctx sdk.Context, chk transfertypes.ChannelKeeper, rlk ratelimitkeeper.Keeper, whitelistedChannelsByID map[string]struct{}) error {
 	logger := ctx.Logger().With("upgrade", UpgradeName)
 	channels := chk.GetAllChannelsWithPortPrefix(ctx, transfertypes.PortID)
 	logger.Info("adding limits to channels", "channels_count", len(channels))
 	for _, ch := range channels {
+		// if there is no whitelisted channel it sets to all the available transfer channels
+		if len(whitelistedChannelsByID) != 0 {
```

Comment:
Case for retrocompatibility (already run upgrade in testnet) which rate limited all the available channels